colossalai/nn/layer/parallel_2d/_operation.py

```diff
@@ -76,7 +77,7 @@ def forward(
         pipeline_parallel_size: int,
         tensor_parallel_size: int,
     ) -> Tensor:
-
+        A = A.clone().detach()
```

Comment:
Do you try `A = A.detach()`?

Reply:
I didn't try it. However, I think `A.detach()` will lead to incorrect computation graph because `A` is the input tensor of the linear layer. If `A` is detached from the graph, the graph will be disconnected.

Reply:
According to the PyTorch documentation, `detach` will create a tensor which shares the storage with the original tensor. If there happens to be some in-place operations in the forward and backward functions, there will be incorrect arithmetic results.

<img width="889" alt="Screenshot 2022-04-11 at 1 40 14 PM" src="https://user-images.githubusercontent.com/31818963/162671889-fe986ad8-74c1-497c-a406-a1c2b21061aa.png">

Reply:
OK